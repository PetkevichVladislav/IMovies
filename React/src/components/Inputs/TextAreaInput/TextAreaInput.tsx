import { FC, useState } from "react";

import "./TextAreaInput.scss";
import "../Input.scss";

export interface ITextAreaInput{
    placeholder: string;
    value?: string;
    height?: string;
    name?: string;
    onChange?: (value: string) => void;
}

export const TextAreaInput : FC<ITextAreaInput> = ({placeholder, value = "", height, name, onChange}) => {    
    return (
        <>
            <textarea style={{height:`${height}`}} 
            className="input text-area-input" 
            name={name} 
            placeholder={placeholder} 
            value={value}
            data-testid="text-area-input"
            onChange={event => onChange && onChange(event.target.value)}/>
        </>
    )
}