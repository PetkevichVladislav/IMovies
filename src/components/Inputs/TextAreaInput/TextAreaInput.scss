.text-area-input{
    width: 100%;
    resize: vertical;
    max-height: 300px;
    min-height: 57px;
    padding: 18px;
}