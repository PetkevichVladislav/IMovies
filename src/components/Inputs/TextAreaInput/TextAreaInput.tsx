import { FC, useState } from "react";

import "./TextAreaInput.scss";
import "../Input.scss";

export interface ITextAreaInput{
    placeholder: string;
    initialValue?: string;
    height?: string;
    name?: string;
}

export const TextAreaInput : FC<ITextAreaInput> = ({placeholder, initialValue = "", height, name}) => {
    const [value, setValue] = useState<string>(initialValue);
    
    const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
        setValue(e.target.value);
    };
    
    return (
        <>
            <textarea style={{height:`${height}`}} 
            className="input text-area-input" 
            name={name} 
            placeholder={placeholder} 
            onChange={handleChange} 
            value={value}
            data-testid="text-area-input"/>
        </>
    )
}