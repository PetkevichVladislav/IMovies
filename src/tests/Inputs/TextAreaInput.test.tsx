import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { ITextAreaInput, TextAreaInput } from '../../components/Inputs/TextAreaInput/TextAreaInput';

describe("Text Area Input", () => {
    const textAreaInputData : ITextAreaInput = {
        placeholder: "input your data",
        value: "text",
        name: "data",
        
    };

    it("When data is valid component should renders correctly", () =>{
        const tree = renderer.create(<TextAreaInput {...textAreaInputData}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("when input is created it should be properly initialized", () =>{
        render(<TextAreaInput {...textAreaInputData}/>);
        const input = screen.getByTestId("text-area-input");
        expect(input).toHaveAttribute("name",textAreaInputData.name);
        expect(input).toHaveAttribute("placeholder",textAreaInputData.placeholder);
    });
});