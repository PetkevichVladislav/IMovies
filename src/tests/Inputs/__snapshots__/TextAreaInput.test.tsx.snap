// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Area Input When data is valid component should renders correctly 1`] = `
<textarea
  className="input text-area-input"
  data-testid="text-area-input"
  name="data"
  onChange={[Function]}
  placeholder="input your data"
  style={
    Object {
      "height": "undefined",
    }
  }
  value="text"
/>
`;
